File: cpp/1423.cpp
#include <iostream>
#include <vector>
#include <numeric>
using namespace std;

class Solution {
public:
    int maxScore(vector<int>& cardPoints, int k) {
        int len = cardPoints.size();
        int cur = accumulate(cardPoints.begin() + len - k, cardPoints.end(), 0);
        int ans = cur;
        for (int i = 0; i < k; ++i) {
            cur = cur - cardPoints[len - k + i] + cardPoints[i];
            if (cur > ans) {
                ans = cur;
            }
        }
        return ans;
    }
};

File: python3/1423.py
from typing import List
class Solution:
    def maxScore(self, cardPoints: List[int], k: int) -> int:
        length = len(cardPoints)
        cur = sum(cardPoints[length - k : length])
        ans = cur
        for i in range(0, k):
            cur = cur - cardPoints[length - k + i] + cardPoints[i]
            if cur > ans:
                ans = cur
        return ans